stdlib/universe/map_vectorize_conditionals_test.flux

```diff
@@ -245,3 +245,21 @@ testcase vec_conditional_null_test_consequent_alternate {
 
     testing.diff(want: want, got: got)
 }
+
+testcase vec_nested_logical_conditional_repro {
+    // Using a logical expr in the test of the conditional expr produced a
+    // runtime error:
+    // ```
+    // cannot use test of type vector in conditional expression; expected boolean
+    // ```
```

Comment:
In theory, I think we might have seen this for any nested expression where logical ops were children of a larger expression.